Sming/SmingCore/Network/WebsocketClient.cpp

```diff
@@ -0,0 +1,422 @@
+/* 
+ * File:   WebsocketClient.cpp
+ * Author: https://github.com/hrsavla
+ *
+ * Created on August 4, 2015, 1:37 PM
+ * This Websocket Client library is ported by me into Sming from  
+ * https://github.com/MORA99/Stokerbot/tree/master/Libraries/WebSocketClient 
+ * 
+ * 
+ */
+
+#include "WebsocketClient.h"
+
+WebsocketClient::WebsocketClient(bool autoDestruct /*= false*/) : TcpClient(autoDestruct)
+{
+    this->Mode = ws_Disconnected;
+}
+
+WebsocketClient::~WebsocketClient()
+{
+    
+}
+/* Function Name: startWS
+ * Description: Start Websocket Client. It connects to Server and initiates 
+ *              handshakes
+ * Parameters: url - Url of the server in format ws://echo.websocket.org
+ *              _callback - Callback function which receives incoming data
+ *              eg. void wsMessageReceived(String message)             
+ */
+bool WebsocketClient::startWS(String url, WebSocketRxCallback _callback /* = NULL*/)
```

Comment:
Better name it in standard way. "connect" is most usual for that case in Sming environment. Another is "begin".
You can look MQTT client as naming example also.